apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConnection.cs

```diff
@@ -889,41 +889,18 @@ private Action GetTransactionEnlister()
         /// with a read-only transaction with the given <see cref="TimestampBound" /> settings
         /// </summary>
         /// <param name="timestampBound">Specifies the timestamp or maximum staleness of a read operation. May be null.</param>
-        public void OpenAsReadOnly(TimestampBound timestampBound = null)
-        {
-            // Note: This has to be checked on the current thread, which is why we don't just use Task.Run
-            // and delegate to OpenAsReadOnlyAsync
-            var transaction = Transaction.Current;
-            if (transaction == null)
-            {
-                throw new InvalidOperationException($"{nameof(OpenAsReadOnlyAsync)} should only be called within a TransactionScope.");
-            }
-            if (!EnlistInTransaction)
-            {
-                throw new InvalidOperationException($"{nameof(OpenAsReadOnlyAsync)} should only be called with ${nameof(EnlistInTransaction)} set to true.");
-            }
-            Open(() => EnlistTransaction(transaction, timestampBound ?? TimestampBound.Strong, transactionId: null, commitDelay: null));
-        }
+        [Obsolete($"Use the {nameof(Open)} that takes a {nameof(AmbientTransactionOptions)} parameter instead.")]
```

Comment:
I like the new methods... I'm not sure about making the existing ones obsolete immediately. (I'm really on the fence.) Let's chat about it.